src/theory/arrays/inference_manager.h

```diff
@@ -45,14 +45,15 @@ class InferenceManager : public TheoryInferenceManager
    * @return true if the fact was successfully asserted, and false if the
    * fact was redundant.
    */
-  bool assertInference(TNode atom, bool polarity, TNode reason, PfRule id);
+  bool assertInference(TNode atom, bool polarity, TNode reason, InferenceId id, PfRule pfr);
```

Comment:
I think our style would do id before reason (InferenceId comes after the conclusion, and before any explanations).